### DISCUSSION TITLE
Allow making `SpinBox` buttons horizontally laid out rather than vertically

### DISCUSSION THREAD

**Discussion by mieldepoche** [godotengine/godot-proposals#9479] [GUI]
Currently, `SpinBox` has two vertically stacked buttons to increment/decrement the value. However, this makes them very small and tedious to aim at, or at least harder than it needs to be.

I have been using inkscape in the past few days, and look at that: taller, easier-to-aim-at buttons!

|`SpinBox` in godot|in inkscape|
|-|-|
|![image](https://github.com/godotengine/godot-proposals/assets/71937306/9045941c-ed50-4a9e-9e8a-359e8d7bd2a9)|![Screenshot from 2024-04-08 16-11-18](https://github.com/godotengine/godot-proposals/assets/71937306/6ea8f087-6d37-4313-862e-e2a37d42672f)|

I prefer those ones a lot. I think they're easier to hit, and in practice I don't think they would eat up precious space: do people ever export ints so large they fill up the line?
The current style could still apply when lacking space or if the `SpinBox` is too small or if the user chooses so, but I believe the editor would benefit from that improvement since most of its spinboxes (inspector and project settings) are usually wide.

I compressed the column a bit in my inspector to make a point:
![image](https://github.com/godotengine/godot-proposals/assets/71937306/9587ca00-682d-4253-95e2-14e8ef0f492f)

I think there's room for another button in there, and 5 digits would fit in a very compressed column.

rough mockup, and size comparison:

![image](https://github.com/godotengine/godot-proposals/assets/71937306/5d811379-a188-486b-96c1-ffc47de5769e)

as you can see, this is sacrificing display space for usability.
Again, I don't know if someone with a tiny monitor ever needed to input a 10-digits value in a spinbox, which is why I believe this would be a good editor usability improvement.

as a side note, using actual buttons nodes would provide the benefit of mouse-hover and click feedback, which is so subtle in godot it might as well not happen, but feels good in inkscape:

https://github.com/godotengine/godot-proposals/assets/71937306/ea0bba15-8aa9-4624-b3a5-2a1888d6e719

**Comment by AThousandShips** [comment]
This should be configurable, rather than changed completely, the ability of `SpinBox` to be used in narrow forms is important

**Comment by Shadowblitz16** [comment]
Also stuff like this...
`< 0 >`
instead of just this...
`0 -+`

**Comment by Mickeon** [comment]
It's not quite the same, but the more styling options of https://github.com/godotengine/godot/pull/89265 may help make this a reality.
